### PR TITLE
Fixing design doc about `DataContractSerializer`, and adding the starting code

### PR DESCRIPTION
**Pull request by cablooshe**

### CODE REVIEW

**Comment by cablooshe** [comment]
@mconnew Wanted to double check that this is good as an update to the info on `DataContractSerializer` in the design doc. Is everything in there accurate?

**Comment by jkotas** [comment]
@cablooshe Could you please rebase away all the code changes from this PR, and keep just the doc update?

File: docs/specs/alcproxies.md [hunk 1]
```diff
@@ -171,10 +167,8 @@ Instead of moving a specific type through a `BinaryFormatter`, using `DataContra
 
 Benefits of `DataContractSerializer`:
 * It may be easier compared to `BinaryFormatter` to cast this effectively across the type-isolation barrier.
-* Adding attributes is similar to `BinaryFormatter`, so any type should be able to make it across the barrier.
-
-Concerns with `DataContractSerializer`:
-* Many similar concerns to `BinaryFormatter`, such as potential security issues and the manual use of [DataContractAttribute] and [DataMemberAttribute] needing to be used. There may also be limits to what types can be serialized using `DataContractSerializer`.
+* There's no need to add attributes if you want the default behavior for `DataContractSerializer` (Serializes all public members).
+* There are none of the security issues compared to `BinaryFormatter` (or at least there are fewer).
```
Comment (mconnew) [inline docs/specs/alcproxies.md]: There are no known security issues with DCS. Fewer is saying > 0 which isn't the case.

Comment (sbomer) [reply docs/specs/alcproxies.md]: How about "It doesn't have the security issues known to exist in `BinaryFormatter`"?

**Pull request merged**